nixos/modules/services/desktops/pipewire/pipewire.nix

```diff
@@ -180,6 +195,12 @@ in {
     systemd.user.sockets.pipewire.wantedBy = lib.mkIf cfg.socketActivation [ "sockets.target" ];
     systemd.user.sockets.pipewire-pulse.wantedBy = lib.mkIf (cfg.socketActivation && cfg.pulse.enable) ["sockets.target"];
 
+    # also depend on the session manager if enabled
+    systemd.services.${cfg.sessionManager.package.unitName}.enable = if cfg.sessionManager.enable then cfg.systemWide else false;
+    systemd.services.${cfg.sessionManager.package.unitName}.wantedBy = [ "pipewire.service" ];
+    systemd.user.services.${cfg.sessionManager.package.unitName}.enable = if cfg.sessionManager.enable then !cfg.systemWide else false;
+    systemd.user.services.${cfg.sessionManager.package.unitName}.wantedBy = [ "pipewire.service" ];
```

Comment:
These conflict with each other, you can only assign `systemd{.user}.services.${cfg.sessionManager.package.unitName}` once so to set both it has to be done with an object rather than standalone attributes, i.e.:
```
systemd.services."${cfg.sessionManager.package.unitName}" = {
  enable = if cfg.sessionManager.enable then cfg.systemWide else false;
  wantedBy = [ "pipewire.service" ];
};
systemd.user.services."${cfg.sessionManager.package.unitName}" = {
  enable = if cfg.sessionManager.enable then !cfg.systemWide else false;
  wantedBy = [ "pipewire.service" ];
};
```

But as mentioned in another comment it's probably better to not do this whole dance at all and just get rid of `service.pipewire.sessionManager` completely. The only thing it gets us atm is it says that two places are trying to set the option, and there is an assertion to guard against that anyway.

Reply:
All right, let's keep it simple for now then.